cmd/create/cluster/cmd.go

```diff
@@ -3610,35 +3626,53 @@ func isValidCidrRange(
 		!serviceNetwork.Contains(subnetIP)
 }
 
-func minReplicaValidator(multiAZ bool, isHostedCP bool, privateSubnetsCount int) interactive.Validator {
+func minReplicaValidator(replicaSizeValidationArgs *machinepool.ReplicaSizeValidationArgs) interactive.Validator {
 	return func(val interface{}) error {
 		minReplicas, err := strconv.Atoi(fmt.Sprintf("%v", val))
 		if err != nil {
 			return err
 		}
 
-		if isHostedCP && minReplicas < 2 {
+		if replicaSizeValidationArgs.IsHostedCp && minReplicas < 2 {
 			return fmt.Errorf("hosted Control Plane clusters require a minimum of 2 nodes, "+
 				"but %d was requested", minReplicas)
 		}
 
-		return clustervalidations.MinReplicasValidator(minReplicas, multiAZ, isHostedCP, privateSubnetsCount)
+		err = machinepool.ValidateClusterVersionWithMaxNodesLimit(
+			replicaSizeValidationArgs.ClusterVersion, minReplicas, replicaSizeValidationArgs.IsHostedCp)
+		if err != nil {
+			return err
+		}
+
+		return clustervalidations.MinReplicasValidator(
+			minReplicas,
+			replicaSizeValidationArgs.MultiAz,
+			replicaSizeValidationArgs.IsHostedCp,
+			replicaSizeValidationArgs.PrivateSubnetsCount,
+		)
 	}
 }
 
-func maxReplicaValidator(multiAZ bool, minReplicas int, isHostedCP bool,
-	privateSubnetsCount int) interactive.Validator {
+func maxReplicaValidator(
+	minReplicas int, replicaSizeValidationArgs *machinepool.ReplicaSizeValidationArgs) interactive.Validator {
```

Comment:
Any reason not to include the min replicas to the validation args?

Reply:
Or maybe have the max replicas validator function be part of the ReplicaSizeValidationArgs struct instead so you don't need to supply it but just apply based on existing instance

Reply:
I can go ahead and add "MinReplicas" an attribute for the "ReplicaSizeValidationArgs" struct.
However, I am personally a bit hesitant with setting "maxReplicaValidator" or "minReplicaValidator" as a method for the struct because, it seems we have two functions, (from different pkg) sharing the same function name but with the same signature. 

https://github.com/openshift/rosa/blob/cas_validations_rosacli/cmd/create/cluster/cmd.go#L3629-L3678
VS.
https://github.com/openshift/rosa/blob/cas_validations_rosacli/pkg/machinepool/helper.go#L202-L241

The two functions have slightly different validation conditions, which I fear may cause some regression issues if I abruptly merge them together. 

A spike should be created as a separate effort to see if min max replica size validations should differ between day 1 and day 2 events. WDYT?

Reply:
I don't think you need to merge it together but just have two methods with different names :thinking: , considering they check different things anyways